sequentialreader.go

```diff
@@ -81,7 +82,13 @@ func (sr *seqReader) readHeaders() {
 
 	er := &errReader{Reader: sr.shp}
 	// shp headers
-	io.CopyN(ioutil.Discard, er, 24)
+	var magic int32
+	binary.Read(er, binary.BigEndian, &magic)
+	if er.e == nil && magic != 9994 {
+		er.e = errors.New("File signature doesn't match shapefile")
```

Comment:
Please do not capitalize error message (https://github.com/golang/go/wiki/CodeReviewComments#error-strings) and avoid contractions. (`don't` -> `do not`)
```suggestion
		er.e = errors.New("file signature does not match shapefile")
```